tests/unit/modules/test_vsphere.py

```diff
@@ -1174,6 +1174,437 @@ def test_returned_value(self):
 
 
 @skipIf(NO_MOCK, NO_MOCK_REASON)
```

Comment:
I tend to wonder if it might be a good idea to mark these tests as destructive. I understand that at present they are fully mocked but my fear here is that somebody is going to end up modifying the module under test, the mocks won't cover all cases and Bad Things Will Happen (tm) to somebody who runs the test.

On the other hand, I'll totally accept that I'm just being crazy and paranoid here. :]

Reply:
Hopefully the person that does the changes, runs the tests and bad things happen to him/her. :)

I personally don't think there should be any destructive unit tests, cause unit tests should not have any side effects on end systems. This being said, I am fine either way. It's very hard to protect against devs that don't change unit tests when changing the base code.

Reply:
Fair enough. :] After giving this some more thought, I think you're right here. Let's go ahead and get this in. Thanks, @alexbleotu